api/weather/api.go

```diff
@@ -1,28 +1,89 @@
 package weather
 
 import (
+	"encoding/json"
 	"fmt"
+	"log"
 	"os"
+	"time"
 
 	"github.com/pisign/pisign-backend/api"
+	"github.com/pisign/pisign-backend/utils"
 )
 
-// API is the function that gets called to get the data from the weather API
+// Data is the function that gets called to get the data from the weather API
 // This should be the only function that gets called from an external service
-func API(data chan string) {
+func (a *API) Data() string {
```

Comment:
Any reason why we are not using the `.Get()` method here? We could change how it gets passed the args because this is essentially the same thing.

Reply:
I'm fine changing it to `Get`. I really didn't integrate our two codebases together, that's the next step. That might be best saved for a separate PR

Reply:
I feel a bit weird pushing stuff to master that is going to be changed so soon. Perhaps instead of merging this into master, we can merge into a separate branch then merge that into master once integration is done?

Reply:
Are you referring to #9? That's a pretty major set of changes that rely on this PR. So I was thinking of merging that after this one. The problem is there are so many big architectural changes in these PRs, it's hard to add any other features that rely on websockets